### PR TITLE
Gripper calib

### PR DESCRIPTION
**Pull request by pinakjani**
Added gripper calibration and finger state calculation for swing arm and rack and pinion mechanism.

### CODE REVIEW

File: src/xs_driver.cpp [hunk 1]
```diff
@@ -870,10 +903,18 @@ bool InterbotixDriverXS::retrieve_motor_configs(
     Gripper gripper;
     // load all info from the single_gripper node into the Griper struct, substituting the default
     //  values if not given the value
+    gripper.type = single_gripper["type"].as<std::string>("swing_arm");
```
Comment (lukeschmitt-tr) [inline src/xs_driver.cpp]: Make `"swing_arm"` and `"rack_and_pinion"` namespaced constant strings like what is done for mode and profile_type variables.

File: src/xs_driver.cpp [hunk 2]
```diff
@@ -748,28 +768,38 @@ float InterbotixDriverXS::convert_linear_position_to_radian(
...
+    float arm_length = gripper_map[name].arm_length;
+    float horn_radius = gripper_map[name].horn_radius;
+    float a1 = horn_radius * sin(angular_position);
+    float c = sqrt(pow(horn_radius, 2) - pow(a1, 2));
+    float a2 = sqrt(pow(arm_length, 2) - pow(c, 2));
+    return a1 + a2;
+  } else {
+    // Conversion in rack and pinion gripper [circumference = r * theta]
+    return gripper_map[name].pitch_radius * angular_position;
+  }
```
Comment (lukeschmitt-tr) [inline src/xs_driver.cpp]: Keep comment style consistent here

**Review by lukeschmitt-tr — CHANGES_REQUESTED** [review]
Very minor changes but looks good to me other than that

**Review by pinakjani — COMMENTED** [review]
Review Done. Resolved Changes.

File: include/interbotix_xs_driver/xs_common.hpp [hunk 3]
```diff
@@ -113,6 +113,14 @@ inline static const std::string TIME = "time";
 
 }  // profile
 
+namespace gripper_type
+{
+
+// constants for command types
```
Comment (lukeschmitt-tr) [inline include/interbotix_xs_driver/xs_common.hpp]:
```suggestion
// constants for gripper types
```

**Review by lukeschmitt-tr — APPROVED** [review]

**Pull request merged**